homeassistant/components/motion_blinds/config_flow.py

```diff
@@ -31,33 +37,71 @@ def __init__(self):
         """Initialize the Motion Blinds flow."""
         self.host = None
         self.key = None
+        self.ips = []
 
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         errors = {}
         if user_input is not None:
-            self.host = user_input[CONF_HOST]
-            self.key = user_input[CONF_API_KEY]
-            return await self.async_step_connect()
+            self.host = user_input.get(CONF_HOST)
+
+            if self.host is not None:
```

Comment:
Do  you need to worry about other cases like empty string? I'm wondering if an "if not self.host:" here is what you need. I haven't dealt with an optional field in config flow yet so I wasn't sure how the data flow handles it.

Reply:
If the user does not enter anything in the Host field `user_input.get(CONF_HOST)` will return None.
Not entering anything in the field is basically entering an empty string, so empty string will also result in None.

If a user enters a wrong IP adress or someting else which is not an IP adress you will get a connection_error with a proper response text.

So I think this is okay.